### PR TITLE
Code reference for test_item

### PR DESCRIPTION
**Pull request by antl31**
Also updated travis ci, , added docstrings

### CODE REVIEW

File: pytest_reportportal/listener.py [hunk 1]
```diff
@@ -1,39 +1,66 @@
...
 
 
 class RPReportListener(object):
+    """RPReportListener class."""
+
     def __init__(self, py_test_service,
                  log_level=logging.NOTSET,
                  endpoint=None):
+        """
+        Initialize RPReport Listener instance.
```
Comment (iivanou) [inline pytest_reportportal/listener.py]:
```
        """Initialize RPReport Listener instance.

        :param py_test_service: PyTestServiceClass instance
        :param log_level:       One of the 'CRITICAL', 'ERROR', 'WARNING','INFO', 
                                'DEBUG', 'NOTSET'
        :param endpoint:        Report Portal API endpoint
        """
```

Please, use the same format for all the docstring.

File: pytest_reportportal/listener.py [hunk 2]
```diff
@@ -1,39 +1,66 @@
...
 
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtest_protocol(self, item):
-        # Adding issues id marks to the test item
-        # if client doesn't support item updates
+        """
+        Adding issues id marks to the test item.
+
+        if client doesn't support item updates
+        :param item:    {name:'',
```
Comment (iivanou) [inline pytest_reportportal/listener.py]: It is an incorrect item description. item is the instance of the Pytest Item class.

File: pytest_reportportal/listener.py [hunk 3]
```diff
@@ -1,39 +1,66 @@
...
+                             endpoint=endpoint)
 
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtest_protocol(self, item):
-        # Adding issues id marks to the test item
-        # if client doesn't support item updates
+        """
+        Adding issues id marks to the test item.
+
+        if client doesn't support item updates
```
Comment (iivanou) [inline pytest_reportportal/listener.py]: I think this sentence is not completed.

File: pytest_reportportal/listener.py [hunk 4]
```diff
@@ -58,6 +85,12 @@ def pytest_runtest_protocol(self, item):
 
     @pytest.hookimpl(hookwrapper=True)
     def pytest_runtest_makereport(self, item):
+        """
+         Change runtest_makereport function.
+
+        :param item: test_item
```
Comment (iivanou) [inline pytest_reportportal/listener.py]: What is test_item?

File: pytest_reportportal/plugin.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+"""Define Pytest built-in functions."""
```
Comment (iivanou) [inline pytest_reportportal/plugin.py]: This module contains/includes ....

File: pytest_reportportal/rp_logging.py [hunk 6]
```diff
@@ -6,14 +8,30 @@
 
 
 class RPLogger(logging.getLoggerClass()):
+    """RPLogger class for logging tests."""
+
     def __init__(self, name, level=0):
+        """
+        Initialize RPLogger instance.
+
+        :param name: name of log
```
Comment (iivanou) [inline pytest_reportportal/rp_logging.py]: Name of log?

File: pytest_reportportal/rp_logging.py [hunk 7]
```diff
@@ -73,6 +101,12 @@ def __init__(self, py_test_service,
         self.endpoint = endpoint
 
     def filter(self, record):
+        """
+        Filter the reportportal_client requests.
```
Comment (iivanou) [inline pytest_reportportal/rp_logging.py]: messages, not requests

File: pytest_reportportal/service.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+"""PyTestServiceClass for work with pytest."""
```
Comment (iivanou) [inline pytest_reportportal/service.py]: This is not a class docstring but module one.

File: pytest_reportportal/service.py [hunk 9]
```diff
@@ -330,6 +400,14 @@ def finish_launch(self, status=None, **kwargs):
         self.rp.finish_launch(**fl_rq)
 
     def post_log(self, message, loglevel='INFO', attachment=None):
+        """
+        Send log.
```
Comment (iivanou) [inline pytest_reportportal/service.py]: Send a log message to the Report Portal.

File: pytest_reportportal/service.py [hunk 10]
```diff
@@ -349,6 +427,11 @@ def post_log(self, message, loglevel='INFO', attachment=None):
         self.rp.log(**sl_rq)
 
     def _stop_if_necessary(self):
+        """
+        Stop tests if errors.
```
Comment (iivanou) [inline pytest_reportportal/service.py]: Stop tests if any error occurs.

File: pytest_reportportal/service.py [hunk 11]
```diff
@@ -390,20 +486,32 @@ def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_pa
...
 
     @staticmethod
-    def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""):
+    def _add_item_hier_parts_parametrize(item, report_parts, tests_parts,
+                                         rp_name=""):
+        """
+        Add item in parents with params.
 
+        :param item: pytest.Item
+        :param report_parts: ''
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: pytest_reportportal/service.py [hunk 12]
```diff
@@ -390,20 +486,32 @@ def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_pa
...
+        :param tests_parts: ''
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: pytest_reportportal/service.py [hunk 13]
```diff
@@ -390,20 +486,32 @@ def _add_item_hier_parts_dirs(item, hier_flag, dirs_level, report_parts, dirs_pa
...
+        :param rp_name: name of report
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: pytest_reportportal/service.py [hunk 14]
```diff
@@ -420,16 +528,33 @@ def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""
...
 
     @staticmethod
-    def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag, report_parts, rp_name=""):
-
+    def _add_item_hier_parts_other(item_parts, item, item_type, hier_flag,
+                                   report_parts, rp_name=""):
+        """
+        Add item to other hierarchy levels.
+
+        :param item_parts: ''
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: pytest_reportportal/service.py [hunk 15]
```diff
@@ -420,16 +528,33 @@ def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""
...
+        :param item: pytest.Item
+        :param item_type: (SUITE, STORY, TEST, SCENARIO, STEP, BEFORE_CLASS,
+         BEFORE_GROUPS, BEFORE_METHOD, BEFORE_SUITE, BEFORE_TEST, AFTER_CLASS,
+        AFTER_GROUPS, AFTER_METHOD, AFTER_SUITE, AFTER_TEST
+        :param hier_flag: ''
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: pytest_reportportal/service.py [hunk 16]
```diff
@@ -420,16 +528,33 @@ def _add_item_hier_parts_parametrize(item, report_parts, tests_parts, rp_name=""
...
+        :param report_parts: ''
```
Comment (iivanou) [inline pytest_reportportal/service.py]: I do not understand this description...

File: tests/test_listener.py [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+    :param mocked_item: Pytest fixture
+    """
+    rp_service = mock.Mock()
+    rp_service.is_item_update_supported = mock.Mock(return_value=False)
+    rp_listener = RPReportListener(rp_service)
+    rp_listener._add_issue_id_marks = mock.Mock()
+
+    next(rp_listener.pytest_runtest_protocol(mocked_item))
+
+    expect(rp_listener._add_issue_id_marks.call_count == 1,
```
Comment (iivanou) [inline tests/test_listener.py]: Why do u use soft assert for just one assert?

File: tests/test_listener.py [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+def test_pytest_runtest_protocol(mocked_item):
+    """Test listener pytest_runtest_protocol hook.
+
+    :param mocked_item: Pytest fixture
+    """
+    rp_service = mock.Mock()
+    rp_service.is_item_update_supported = mock.Mock(return_value=False)
+    rp_listener = RPReportListener(rp_service)
+    rp_listener._add_issue_id_marks = mock.Mock()
```
Comment (iivanou) [inline tests/test_listener.py]: U can use already created fixtures.

File: pytest_reportportal/errors.py [hunk 19]
```diff
@@ -8,4 +8,5 @@ class PytestWarning(UserWarning):
     up to 3.8.0. Get rid of this code once we drop support for Pytest versions
     below 3.8.0.
     """
+
```
[the remark this answers is not in the document]
Comment (antl31) [reply pytest_reportportal/errors.py]: Sure,  D204: 1 blank line required after class docstring

[the remark this answers is not in the document]
Comment (iivanou) [reply pytest_reportportal/errors.py]: I meant that "pass"  can be omitted along with the blank line.

**Pull request closed without merging**